### PR TITLE
Fixing Tx Query issues

### PR DESCRIPTION
**Pull request by crussell-figure**
## Description

* Adding more indicies due to how the queries are being joined and searched
* Adding tx/msg checks to block look up and tx look up
  * Allows for missed txs/msgs to be added under the hood

closes: #71

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/provenance-io/explorer/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Wrote unit and integration
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -34,9 +34,19 @@ Ref: https://keepachangelog.com/en/1.0.0/
 
 ## Unreleased
 
+### Improvements
+* On asset list and asset detail, the supply object now contains initial and circulation rather than circulation and total #52
+* All coin objects (with a balance and a denom) now have a `queryDenom` value to be used for querying based on that denom #52
+* `/api/v2/assets/{id}/holders` now returns a paginated list #52
+* Asset supply total comes from bank module, and supply on asset detail object is now labeled as initial supply #52
```
Comment (channa-figure) [inline CHANGELOG.md:41]: Not sure if you need a comma here...

File: service/src/main/kotlin/io/provenance/explorer/service/async/AsyncCaching.kt [hunk 2]
```diff
@@ -47,12 +48,12 @@ class AsyncCaching(
...
 
-    fun getBlock(blockHeight: Int) = getBlockByHeightFromCache(blockHeight)
-
-    fun getBlockByHeightFromCache(blockHeight: Int) = transaction {
+    fun getBlock(blockHeight: Int, checkTxs: Boolean = false) = transaction {
         BlockCacheRecord.findById(blockHeight)?.also {
             BlockCacheRecord.updateHitCount(blockHeight)
-        }?.block ?: saveBlockEtc(blockService.getBlockAtHeightFromChain(blockHeight))
+        }?.block?.also {
+            if (checkTxs) addTxsToCache(it.block.height(), it.block.data.txsCount, it.block.header.time)
```
Comment (channa-figure) [inline service/src/main/kotlin/io/provenance/explorer/service/async/AsyncCaching.kt:55]: Would it be more efficient to have checkTxs && it.block.data.txCount > 0?

**Review by channa-figure — CHANGES_REQUESTED** [review]
Quick change log fix, if you believe it is needed.  There is also another comment about a checking txs at a block height.  It could maybe bypass a database call if the block tx count is zero.

File: CHANGELOG.md (hunk 1 above)
Comment (crussell-figure) [reply CHANGELOG.md]: Reworded to make it a bit more clear

File: service/src/main/kotlin/io/provenance/explorer/service/async/AsyncCaching.kt (hunk 2 above)
Comment (crussell-figure) [reply service/src/main/kotlin/io/provenance/explorer/service/async/AsyncCaching.kt]: Agreed. Updated

**Pull request merged**